hdk/jack_analyzer/syntax.py

```diff
@@ -0,0 +1,316 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import TypeAlias
+from xml.dom.minidom import Document, Element
+from enum import Enum
+
+
+def _add_child(element: Element, tag_name: str, value: str | None) -> Element:
+    dom_tree = element.ownerDocument
+    child = dom_tree.createElement(tag_name)
+    child.appendChild(dom_tree.createTextNode(value))
+    element.appendChild(child)
+    return child
+
+
+@dataclass(frozen=True)
+class ParameterList:
+    parameters: list[tuple[str, str]]
```

Comment:
Instead of using list[tuple[str, str]], create new dataclass Parameter with attributes type_, var_name.